deps/yoml/yoml-parser.h

```diff
@@ -297,25 +303,77 @@ static inline int yoml__resolve_alias(yoml_t **target, yoml_t *doc, yaml_parser_
     return 0;
 }
 
-static inline yoml_t *yoml_parse_document(yaml_parser_t *parser, yaml_event_type_t *unhandled,
-                                          void *(*mem_set)(void *, int, size_t), const char *filename)
+static inline int yoml__resolve_tag(yoml_t **target, yaml_parser_t *parser, yoml_parse_args_t *parse_args)
+{
+    size_t i;
+
+    if (parse_args->resolve_tag.cb == NULL)
+        return 0;
+
+    if ((*target)->tag != NULL) {
+        yoml_t *resolved = parse_args->resolve_tag.cb((*target)->tag, *target, parse_args->resolve_tag.cb_arg);
+        if (resolved == NULL) {
+            if (parser != NULL) {
+                parser->problem = "could not resolve tag";
+                parser->problem_mark.line = (*target)->line;
+                parser->problem_mark.column = (*target)->column;
+            }
+            return -1;
+        }
+        if (resolved != *target) {
```

Comment:
I do not think you would need this check.

I believe that the callback for tag resolution should return a node with its `_refcnt` incremented, if it decides to return the same node, since it would essentially mean that the callback is _duplicating_ the node being passed in.